bool allow_to_run

